account_statement_completion_compassion/model/completion_rules.py

```diff
@@ -358,3 +359,67 @@ def _generate_invoice_line(self, cr, uid, invoice_id, product, st_line,
             cr, uid, inv_line_data, context=context)
 
         return res
+
+    def _search_partner_by_bvr_ref(self, cr, uid, bvr_ref, context=None,
+                                   search_old_invoices=False):
+        """ Finds a partner given its bvr reference. """
+        partner = None
+        contract_group_obj = self.pool.get('recurring.contract.group')
+        contract_group_ids = contract_group_obj.search(
+            cr, uid, [('bvr_reference', '=', bvr_ref)], context=context)
+        if contract_group_ids:
+            partner = contract_group_obj.browse(
+                cr, uid, contract_group_ids, context=context)[0].partner_id
+        else:
+            # Search open Customer Invoices (with field 'bvr_reference' set)
+            invoice_obj = self.pool.get('account.invoice')
+            invoice_search = [
+                ('bvr_reference', '=', bvr_ref),
+                ('state', '=', 'open')]
+            if search_old_invoices:
+                invoice_search[1] = ('state', 'in', ('open', 'cancel',
+                                                     'paid'))
+            invoice_ids = invoice_obj.search(cr, uid, invoice_search,
+                                             context=context)
+            if not invoice_ids:
+                # Search open Supplier Invoices (with field 'reference_type'
+                # set to BVR)
+                invoice_ids = invoice_obj.search(cr, uid, [
+                    ('reference_type', '=', 'bvr'),
+                    ('reference', '=', bvr_ref),
+                    ('state', '=', 'open')], context=context)
+            if invoice_ids:
+                partner = invoice_obj.browse(
+                    cr, uid, invoice_ids, context=context)[0].partner_id
+        return partner
+
+    def get_sponsor_name(self, cr, uid, st_line, context=None):
+        res = dict()
+        label = st_line['label']
+
+        # Fix encoding problem
+        reload(sys)
```

Comment:
It's a trick to change encoding to UTF8 and avoid some errors with accented characters.

http://stackoverflow.com/questions/2276200/changing-default-encoding-of-python

http://stackoverflow.com/questions/3828723/why-we-need-sys-setdefaultencodingutf-8-in-a-py-script